<launch>
  <param name="/use_sim_time" value="true" />
 <arg name="robot1_init_pose" default="-x 1.5 -y -2 -z 0" />
 <arg name="world_file"  value = "$(find ROS_opencv_improcess)/worlds/newTest.world"/>
 <arg name="gui" default="true"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="gui" value="$(arg gui)" />
    <arg name="world_name" value="$(arg world_file)"/>
  </include>

<!-- include our robots -->
  <include file="$(find ROS_opencv_improcess)/launch/robot.launch">
  <arg name="init_pose_1" value="$(arg robot1_init_pose)" />
  </include>

<node pkg="turtlebot_teleop" type="turtlebot_teleop_key" name="turtlebot_teleop_keyboard" launch-prefix="xterm -e" output="screen">
    <param name="scale_linear" value="0.5" type="double"/>
    <param name="scale_angular" value="1.5" type="double"/>
    <remap from="turtlebot_teleop_keyboard/cmd_vel" to="/robot2/cmd_vel_mux/input/teleop"/>
  </node>
  
<node pkg="ROS_opencv_improcess" type="visionTurtlebot" name="followRobot" output="screen" launch-prefix="xterm -e" >
<remap from="camera/rgb/image_raw" to ="/robot1/camera/rgb/image_raw" />
<remap from="/scan" to ="/robot1/scan" />
<remap from="/cmd_vel_mux/input/teleop" to="robot1/cmd_vel_mux/input/teleop" />
</node>
</launch>
